PMVP/KeyedSubjectHolder.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  KeyedSubjectHolder.swift
+//  PMVP
+//
+//  Created by Aubrey Goodman on 4/23/19.
+//  Copyright © 2019 Aubrey Goodman. All rights reserved.
+//
+
+import RxSwift
+
+open class KeyedSubjectHolder<K: Hashable, T: Proxy<K>> {
+
+	private var subjectMap: [K: BehaviorSubject<T?>] = [:]
+
+	private let scheduler: SchedulerType
+
+	private let preloader: (K, BehaviorSubject<T?>) -> Void
+
+	public init(scheduler: SchedulerType, preloader: @escaping (K, BehaviorSubject<T?>) -> Void = { _,_ in }) {
+		self.scheduler = scheduler
+		self.preloader = preloader
+	}
+
+	public final func subject(for key: K) -> Observable<T?> {
+		return findOrCreateSubject(for: key)
+	}
+
+	public final func notify(_ object: T) {
```

Comment:
This feels like it should have the key in the interface, since the key isn't necessarily the object's key. This change would eliminate the need to use Proxy<K> in the generic list.